packages/xo-server/src/xo-mixins/remotes.js

```diff
@@ -83,7 +83,18 @@ export default class {
 
   async testRemote(remote) {
     const handler = await this.getRemoteHandler(remote)
-    return handler.test()
+    const answer = await handler.test()
+    const { success, ...benchmark } = answer
+
+    if (success) {
+      const benchmarks =
+        remote.benchmarks !== undefined
+          ? [...remote.benchmarks, benchmark]
+          : [benchmark]
+      await this.updateRemote(remote, { benchmarks })
```

Comment:
IMHO, better use directly the private method. Because `benchmarks` should only be modified by the method `testRemote`.